### PR TITLE
More pause options

### PR DESCRIPTION
**Pull request by Henrybk**
This pull request add the possibility of pausing automacro checking and/or the running macro, as two different things. I also added the corresponding console commands. Needs testing.

### CODE REVIEW

File: eventMacro.pl [hunk 1]
```diff
@@ -92,10 +92,13 @@ sub commandHandler {
 		message 
 			"eventMacro MACRO: run macro MACRO\n".
 			"eventMacro list: list available macros\n".
-			"eventMacro status: shows current status\n".
-			"eventMacro stop: stop current macro\n".
-			"eventMacro pause: interrupt current macro\n".
-			"eventMacro resume: resume interrupted macro\n".
+			"eventMacro macro_status: shows current macro status\n".
```
Comment (allanon) [inline eventMacro.pl]: These names are too long to type. It's fine to assume people have an IDE and will be able to autocomplete long variable names, but these are going to be typed by end users. They should be short and easy to remember. I vote to keep the original names for the `macro` set (ie, `status`, `stop`, `pause`, `resume`).

File: eventMacro/Core.pm [hunk 2]
```diff
@@ -70,24 +68,22 @@ sub clean_hooks {
...
-	return ( $self->{Status} ? 1 : 0 );
-}
-
-sub pause {
-	my ($self) = @_;
-	$self->{Paused} = PAUSED;
+sub set_automacro_checking_status {
+	my ($self, $status) = @_;
+	
+	if ($self->{Automacros_Checking_Status} == CHECKING_AUTOMACROS) {
```
Comment (allanon) [inline eventMacro/Core.pm:78]: What happens if `$self->{Automacros_Checking_Status} == $status`?

File: eventMacro/Core.pm [hunk 3]
```diff
@@ -70,24 +68,22 @@ sub clean_hooks {
...
+		Plugins::delHook($self->{AI_pre_Hook_Handle});
+		$self->{AI_pre_Hook_Handle} = undef;
+	} elsif ($status == CHECKING_AUTOMACROS) {
+		$self->{AI_pre_Hook_Handle} = Plugins::addHook( 'AI_pre', sub { $self->AI_pre_checker(); }, undef );
```
Comment (allanon) [inline eventMacro/Core.pm:84]: The trailing `undef` is unnecessary.

File: eventMacro.pl [hunk 4]
```diff
@@ -134,15 +137,75 @@ sub commandHandler {
...
+			message "Automacros are not being checked because there's an uninterruptible macro running ('".$eventMacro->{Macro_Runner}->get_name()."').\n";
+		} else {
+			message "Automacros are not being checked because the user stopped it.\n";
+		}
+	### parameter: automacro_checking_pause
+	} elsif ($arg eq 'automacro_checking_pause') {
+		my $status = $eventMacro->get_automacro_checking_status();
+		if ($status == 0) {
+			message "Automacros won't be checked anymore.\n";
+			$eventMacro->set_automacro_checking_status(2);
```
Comment (allanon) [inline eventMacro.pl]: Should we use the names instead? Eg, `eventMacro::Core::PAUSED_BY_USER`.

File: eventMacro.pl [hunk 5]
```diff
@@ -92,10 +92,13 @@ sub commandHandler {
...
-			"eventMacro status: shows current status\n".
-			"eventMacro stop: stop current macro\n".
-			"eventMacro pause: interrupt current macro\n".
-			"eventMacro resume: resume interrupted macro\n".
+			"eventMacro macro_status: shows current macro status\n".
+			"eventMacro macro_stop: stop current macro\n".
+			"eventMacro macro_pause: pauses current macro execution\n".
+			"eventMacro macro_resume: resumes interrupted macro\n".
+			"eventMacro automacro_checking_status: shows current automacro checking status\n".
+			"eventMacro automacro_checking_pause: pauses automacro checking\n".
```
Comment (allanon) [inline eventMacro.pl]: Maybe make this `eventMacro disable automacros`? That sounds a little better (to me).

File: eventMacro/Core.pm (hunk 2 above)
Comment (Henrybk) [inline eventMacro/Core.pm:78]: There was a return for this case in the previous commit, I removed it by accident.

File: eventMacro.pl (hunk 4 above)
Comment (Henrybk) [inline eventMacro.pl]: Good idea

File: eventMacro.pl (hunk 1 above)
Comment (Henrybk) [inline eventMacro.pl]: I like 'macro_status|stop|pause|resume' but I also think that the automacro commands got way bigger than I would like them to be. Maybe 'automacro_pause|resume|status'?

File: eventMacro.pl [hunk 6]
```diff
@@ -188,10 +251,14 @@ sub commandHandler {
 
 		if ( defined $eventMacro->{Macro_Runner} ) {
 			if ( defined $opt->{override_ai} ) { $eventMacro->{Macro_Runner}->overrideAI( 1 ); }
-			if ( defined $opt->{exclusive} )   { $eventMacro->{Macro_Runner}->interruptible( 0 ); }
+			if ( defined $opt->{exclusive} )   {
+				$eventMacro->{Macro_Runner}->interruptible( 0 );
+				message "[eventMacro] Calling uninterruptible macro '".$arg."'. Automacro checking will be paused until it ends.\n";
```
Comment (allanon) [inline eventMacro.pl]: Should this be a `debug` instead?

**Comment by Henrybk** [comment]
Here is a macro to test the added features in inheritance: http://pastebin.com/FKiYrwzy

To better see the changes it's a good idea to add a message or debug to eventMacro::Core::AI_pre_checker

**Pull request merged**